app/lib/reports_service.rb

```diff
@@ -60,6 +60,29 @@ def self.study_data
         study_hash[study_id][:has_raw_counts] ||= is_raw_counts.present? ? is_raw_counts['is_raw_counts'] : false
       end
     end
+
+    history_hash = ReportsService.study_histories
+    study_hash.keys.each do |study_id|
+      study_hash[study_id][:last_public] = history_hash[study_id] ? history_hash[study_id][:last_public] : nil
```

Comment:
A simpler way to do both of these - it will gracefully handle the non-existent key and default to `nil`:
```suggestion
      study_hash[study_id][:last_public] = history_hash.dig(study_id, :last_public)
      study_hash[study_id][:last_initialized] = history_hash.dig(study_id, :last_initialized)
```